src/bf_dpu_update.py

```diff
@@ -872,7 +956,14 @@ def update_bios(self):
             raise Err_Exception(Err_Num.FAILED_TO_ENABLE_BMC_RSHIM, 'Please make sure rshim on Host side is disabled')
 
         protocol, task_handle = self.update_bfb()
-        self._wait_task(task_handle, max_second=20*60, check_step=2)
+
+        def err_handler(task_state):
+            if protocol == "SCP" and "Please provide server's public key using PublicKeyExchange" in task_state['message']:
+                raise Err_Exception(Err_Num.PUBLIC_KEY_NOT_EXCHANGED)
+            elif protocol == "HTTP" and "Check and restart server's web service" in task_state['message']:
+                raise Err_Exception(Err_Num.HTTP_FILE_SERVER_NOT_ACCESSIBLE, "Server address: {}:{}".format(self._format_ip(self._get_local_ip()), self._local_http_server_port))
+
```

Comment:
do we have other error cases?
If have, let's add else and exception.

Reply:
this is a specific error handler for  only BFB simper update.
The _wait_task will handle other general errors.